### DISCUSSION TITLE
Can't get my Eraser Button to work

### DISCUSSION THREAD

**Discussion by ethanbowering24** [xournalpp/xournalpp#3010] [General]
![image](https://user-images.githubusercontent.com/61316561/113607194-edeb4d80-9616-11eb-9b14-6981a2810cbd.png)

Above is a screenshot of a wacom utility on my system. My laptop stylus has two buttons, one that seems to be recognized as a regular mouse input and the other is treated as an "eraser tip" in xournal, although you just use it by pressing the button and continuing to use the pen as normal. I can not get this functionality to wourk in xournal++. Any ideas?

Below is my xinput --list print:
```
[ebowering@Thinkpad-X1-Yoga-Fedora]~% xinput --list
⎡ Virtual core pointer                          id=2    [master pointer  (3)]
⎜   ↳ Virtual core XTEST pointer                id=4    [slave  pointer  (2)]
⎜   ↳ Wacom Pen and multitouch sensor Finger touch      id=9    [slave  pointer  (2)]
⎜   ↳ Wacom Pen and multitouch sensor Pen stylus        id=10   [slave  pointer  (2)]
⎜   ↳ SynPS/2 Synaptics TouchPad                id=13   [slave  pointer  (2)]
⎜   ↳ TPPS/2 Elan TrackPoint                    id=14   [slave  pointer  (2)]
⎜   ↳ Wacom Pen and multitouch sensor Pen eraser        id=16   [slave  pointer  (2)]
⎣ Virtual core keyboard                         id=3    [master keyboard (2)]
    ↳ Virtual core XTEST keyboard               id=5    [slave  keyboard (3)]
    ↳ Power Button                              id=6    [slave  keyboard (3)]
    ↳ Video Bus                                 id=7    [slave  keyboard (3)]
    ↳ Sleep Button                              id=8    [slave  keyboard (3)]
    ↳ Integrated Camera: Integrated C           id=11   [slave  keyboard (3)]
    ↳ AT Translated Set 2 keyboard              id=12   [slave  keyboard (3)]
    ↳ ThinkPad Extra Buttons                    id=15   [slave  keyboard (3)]
[ebowering@Thinkpad-X1-Yoga-Fedora]~% 
```

**Comment by rolandlo** [comment]
So you are on Fedora? What version of Xournal++ do you use? What are your input system and stylus settings in menu `Edit -> Preferences`?

**Reply by ethanbowering24** [reply]
![image](https://user-images.githubusercontent.com/61316561/113613202-e29c2000-961e-11eb-9ac3-cf24e6a60dae.png)

![image](https://user-images.githubusercontent.com/61316561/113613266-f9427700-961e-11eb-9523-de52986dfc2c.png)

Version 1.0.20. I have tried tinkering with those settings but can never seem to get it to work.

**Comment by rolandlo** [comment]
In the stylus panel you should assign some tools to the buttons. Otherwise pressing the button won't do anything special.
It's strange that no devices have been found. I would advise you to try with Xournal++ version 1.1.0+dev from the [nightly release](https://github.com/xournalpp/xournalpp/releases/tag/nightly)
